examples/shadow-tests.dx

```diff
@@ -59,3 +59,21 @@ x = 1
    (_, _, w) = (1,2,4)
    w
 > 4
+
+-- Testing data shadowing
+data Shadow = Shadow
+> Error: variable already defined: Shadow
+
+data Shadow =
+  Shadow1 Int
+  Shadow1
+> Error: variable already defined: Shadow1
+
+ShadowCon = 1
+data Shadow = ShadowCon
+> Error: variable already defined: ShadowCon
+
+-- In the opposite direction, it already worked properly
```

Comment:
nit: since someone reading the test might not know about the history, "already" is a bit vague. Personally I'd just mention it in the commit message but not as a comment.

Reply:
Agreed, let's remove the comment

Reply:
Completely right, thanks for making the change @apaszke :+1: !